src/Core/Backend.Core.DDD.Tests/Registration/QueryHandlerRegistrationTests.cs

```diff
@@ -134,5 +135,37 @@ public void GivenDuplicatedQueryHandler_WhenAddAllQueryHandlerCalled_ThenBothAre
                 deleteAccountHandlers.Should().Contain(x => x is DuplicatedGetMainAccountQueryHandler);
             }
         }
+        private IServiceCollection Collection { get; } = new ServiceCollection();
+        [Fact]
+        public void CanRegisterAllQueriesWithFromApplicationDependencies()
+        {
+            Collection.Scan(scan => scan
+                .FromApplicationDependencies()
+                .AddClasses(classes => classes.AssignableTo(typeof(IQueryHandler<,>)))
+                .AsSelfWithInterfaces()
+            );
+            
+
+            Assert.Collection(Collection,
+                t => Assert.Equal(typeof(IQueryHandler<GetAccountList, IReadOnlyCollection<Account>>), t.ServiceType)
```

Comment:
@oskardudycz should we look for the IQueryHandler interface with two generic parameters to make the valid test for this case ?